currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -160,7 +157,8 @@ def _update(self, date_from, date_to, newest_only=False):
 
             if not data:
                 if is_scheduled:
-                    provider._schedule_next_run()
+                    # Try again if there is no data yet
+                    provider.next_run = fields.Date.today()
```

Comment:
Isn't the next_run already in the past?

Reply:
Use case:
- Next run: 2019-01-01
- We run cron and NO data is returned. Next run: today
(the idea is to retry again according to the frequency set in cron)

Another possibility would be NOT to update next run, what do you think?

Reply:
We shouldn't IMO. Use case:

- Next run: 2022-12-01, Thursday
- We run cron on 2022-12-02 and the webservice is not available.
- Putting next run = 2022-12-02 will lose data for 2022-12-01

Or am I wrong?

Reply:
Ok, you are right, it is already changed.